web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -1069,38 +1069,58 @@
                   scope.popupid = '#linkto' + scope.mode + '-popup';
                   scope.alertMsg = null;
                   scope.layerSelectionMode = 'multiple';
+                  scope.onlineSrcLink = '';
+                  scope.addOnlineSrcInDataset = true;
 
                   gnOnlinesrc.register(scope.mode, function() {
                     $(scope.popupid).modal('show');
 
                     // parameters of the online resource form
                     scope.srcParams = {selectedLayers: []};
 
-                    var searchParams = {
-                      type: scope.mode
-                    };
+                    var searchParams = {};
+                    if (scope.mode === 'service') {
+                      searchParams.type = scope.mode;
+                    } else {
+                      // Any records which are not services
+                      // ie. dataset, series, ...
+                      searchParams['without-type'] = 'service';
```

Comment:
What is this used for? Searching in the code for `without-type` this seem the only place used.

Reply:
"Also fix the search which was only listing datasets (and for example not series) - replaced by search for all which is not a service."

When you are in a service and click on link dataset, then the search was listing all records with type = dataset which means that series were not listed. Which was not really expected by users (eg for INSPIRE series are supposed to be linked to services). So now, the search list all records which are not services.

"without-" comes from https://github.com/geonetwork/core-geonetwork/pull/3683 which allows now to do negative queries